### PR TITLE
INTMDB-313: Update the project resource with new settings

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

Update the project resource with new settings feature

Link to any related issue(s):
https://jira.mongodb.org/browse/INTMDB-313

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/data_source_mongodbatlas_project.go [hunk 1]
```diff
@@ -81,6 +81,26 @@ func dataSourceMongoDBAtlasProject() *schema.Resource {
 					},
 				},
 			},
+			"is_collect_database_specifics_statistics_enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_project.go:87]: [q] Could you update the tests with these new fields?

Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_project.go:87]: Tests added

File: mongodbatlas/resource_mongodbatlas_project.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 	"errors"
 	"log"
 	"net/http"
+
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_project.go]:
```suggestion
```

File: mongodbatlas/data_source_mongodbatlas_project.go (hunk 1 above)
Comment (andreaangiolillo) [reply mongodbatlas/data_source_mongodbatlas_project.go:87]: Could you show me where? I do not see them

Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_project.go:87]: Apologies SDK had these will add to datasource

File: website/docs/d/project.html.markdown [hunk 3]
```diff
@@ -96,4 +96,11 @@ The following are valid roles:
   * `GROUP_DATA_ACCESS_READ_ONLY`
   * `GROUP_CLUSTER_MANAGER`
 
+
+* `is_collect_database_specifics_statistics_enabled` - Flag that indicates whether to enable statistics in cluster metrics collection for the project.
```
Comment (themantissa) [inline website/docs/d/project.html.markdown]:
```suggestion
* `is_collect_database_specifics_statistics_enabled` - Flag that indicates whether to enable statistics in [cluster metrics](https://www.mongodb.com/docs/atlas/monitor-cluster-metrics/) collection for the project.
```

File: website/docs/d/project.html.markdown [hunk 4]
```diff
@@ -96,4 +96,11 @@ The following are valid roles:
...
+* `is_data_explorer_enabled` - Flag that indicates whether to enable Data Explorer for the project. If enabled, you can query your database with an easy to use interface.
+* `is_performance_advisor_enabled` - Flag that indicates whether to enable Performance Advisor and Profiler for the project. If enabled, you can analyze database logs to recommend performance improvements.
+* `is_realtime_performance_panel_enabled` - Flag that indicates whether to enable Real Time Performance Panel for the project. If enabled, you can see real time metrics from your MongoDB database.
+* `is_schema_advisor_enabled` - Flag that indicates whether to enable Schema Advisor for the project. If enabled, you receive customized recommendations to optimize your data model and enhance performance. Disable this setting to disable schema suggestions in the Performance Advisor and the Data Explorer.
```
Comment (themantissa) [inline website/docs/d/project.html.markdown]:
```suggestion
* `is_schema_advisor_enabled` - Flag that indicates whether to enable Schema Advisor for the project. If enabled, you receive customized recommendations to optimize your data model and enhance performance. Disable this setting to disable schema suggestions in the [Performance Advisor](https://www.mongodb.com/docs/atlas/performance-advisor/#std-label-performance-advisor) and the [Data Explorer](https://www.mongodb.com/docs/atlas/atlas-ui/#std-label-atlas-ui).
```

File: website/docs/d/projects.html.markdown [hunk 5]
```diff
@@ -76,5 +76,9 @@ The following are valid roles:
   * `GROUP_DATA_ACCESS_READ_ONLY`
   * `GROUP_CLUSTER_MANAGER`
 
-
+* `is_collect_database_specifics_statistics_enabled` - Flag that indicates whether to enable statistics in cluster metrics collection for the project.
+* `is_data_explorer_enabled` - Flag that indicates whether to enable Data Explorer for the project. If enabled, you can query your database with an easy to use interface.
+* `is_performance_advisor_enabled` - Flag that indicates whether to enable Performance Advisor and Profiler for the project. If enabled, you can analyze database logs to recommend performance improvements.
+* `is_realtime_performance_panel_enabled` - Flag that indicates whether to enable Real Time Performance Panel for the project. If enabled, you can see real time metrics from your MongoDB database.
+* `is_schema_advisor_enabled` - Flag that indicates whether to enable Schema Advisor for the project. If enabled, you receive customized recommendations to optimize your data model and enhance performance. Disable this setting to disable schema suggestions in the Performance Advisor and the Data Explorer.
```
Comment (themantissa) [inline website/docs/d/projects.html.markdown]:
```suggestion
* `is_schema_advisor_enabled` - Flag that indicates whether to enable Schema Advisor for the project. If enabled, you receive customized recommendations to optimize your data model and enhance performance. Disable this setting to disable schema suggestions in the [Performance Advisor](https://www.mongodb.com/docs/atlas/performance-advisor/#std-label-performance-advisor) and the [Data Explorer](https://www.mongodb.com/docs/atlas/atlas-ui/#std-label-atlas-ui).
```

File: website/docs/d/projects.html.markdown [hunk 6]
```diff
@@ -76,5 +76,9 @@ The following are valid roles:
   * `GROUP_DATA_ACCESS_READ_ONLY`
   * `GROUP_CLUSTER_MANAGER`
 
-
+* `is_collect_database_specifics_statistics_enabled` - Flag that indicates whether to enable statistics in cluster metrics collection for the project.
```
Comment (themantissa) [inline website/docs/d/projects.html.markdown]:
```suggestion
* `is_collect_database_specifics_statistics_enabled` - Flag that indicates whether to enable statistics in [cluster metrics](https://www.mongodb.com/docs/atlas/monitor-cluster-metrics/) collection for the project.
```

File: website/docs/r/project.html.markdown [hunk 7]
```diff
@@ -74,8 +80,14 @@ api_keys allows one to assign an existing organization programmatic API key to a
...
   * `GROUP_DATA_ACCESS_READ_ONLY`
-  * `GROUP_CLUSTER_MANAGER`
-
+  * `GROUP_CLUSTER_MANAGER`  
+ 
+* `is_collect_database_specifics_statistics_enabled` - (Optional) Flag that indicates whether to enable statistics in cluster metrics collection for the project.
+* `is_data_explorer_enabled` - (Optional) Flag that indicates whether to enable Data Explorer for the project. If enabled, you can query your database with an easy to use interface.
+* `is_performance_advisor_enabled` - (Optional) Flag that indicates whether to enable Performance Advisor and Profiler for the project. If enabled, you can analyze database logs to recommend performance improvements.
+* `is_realtime_performance_panel_enabled` - (Optional) Flag that indicates whether to enable Real Time Performance Panel for the project. If enabled, you can see real time metrics from your MongoDB database.
+* `is_schema_advisor_enabled` - (Optional) Flag that indicates whether to enable Schema Advisor for the project. If enabled, you receive customized recommendations to optimize your data model and enhance performance. Disable this setting to disable schema suggestions in the Performance Advisor and the Data Explorer.
```
Comment (themantissa) [inline website/docs/r/project.html.markdown]:
```suggestion
* `is_schema_advisor_enabled` - (Optional) Flag that indicates whether to enable Schema Advisor for the project. If enabled, you receive customized recommendations to optimize your data model and enhance performance. Disable this setting to disable schema suggestions in the [Performance Advisor](https://www.mongodb.com/docs/atlas/performance-advisor/#std-label-performance-advisor) and the [Data Explorer](https://www.mongodb.com/docs/atlas/atlas-ui/#std-label-atlas-ui).
```

File: website/docs/r/project.html.markdown [hunk 8]
```diff
@@ -74,8 +80,14 @@ api_keys allows one to assign an existing organization programmatic API key to a
   * `GROUP_DATA_ACCESS_ADMIN`
   * `GROUP_DATA_ACCESS_READ_WRITE`
   * `GROUP_DATA_ACCESS_READ_ONLY`
-  * `GROUP_CLUSTER_MANAGER`
-
+  * `GROUP_CLUSTER_MANAGER`  
+ 
+* `is_collect_database_specifics_statistics_enabled` - (Optional) Flag that indicates whether to enable statistics in cluster metrics collection for the project.
+* `is_data_explorer_enabled` - (Optional) Flag that indicates whether to enable Data Explorer for the project. If enabled, you can query your database with an easy to use interface.
```
Comment (themantissa) [inline website/docs/r/project.html.markdown]:
```suggestion
* `is_data_explorer_enabled` - (Optional) Flag that indicates whether to enable Data Explorer for the project. If enabled, you can query your database with an easy to use interface.  When Data Explorer is disabled, you cannot terminate slow operations from the [Real-Time Performance Panel](https://www.mongodb.com/docs/atlas/real-time-performance-panel/#std-label-real-time-metrics-status-tab) or create indexes from the [Performance Advisor](https://www.mongodb.com/docs/atlas/performance-advisor/#std-label-performance-advisor). You can still view Performance Advisor recommendations, but you must create those indexes from [mongosh](https://www.mongodb.com/docs/mongodb-shell/#mongodb-binary-bin.mongosh).
```

File: website/docs/r/project.html.markdown [hunk 9]
```diff
@@ -74,8 +80,14 @@ api_keys allows one to assign an existing organization programmatic API key to a
   * `GROUP_DATA_ACCESS_ADMIN`
   * `GROUP_DATA_ACCESS_READ_WRITE`
   * `GROUP_DATA_ACCESS_READ_ONLY`
-  * `GROUP_CLUSTER_MANAGER`
-
+  * `GROUP_CLUSTER_MANAGER`  
+ 
+* `is_collect_database_specifics_statistics_enabled` - (Optional) Flag that indicates whether to enable statistics in cluster metrics collection for the project.
```
Comment (themantissa) [inline website/docs/r/project.html.markdown]:
```suggestion
* `is_collect_database_specifics_statistics_enabled` - (Optional) Flag that indicates whether to enable statistics in [cluster metrics](https://www.mongodb.com/docs/atlas/monitor-cluster-metrics/) collection for the project.
```

**Review by themantissa — CHANGES_REQUESTED** [review]
Just some docs changes.  Made suggestions for all so hopefully not too painful.  TY!

File: website/docs/r/project.html.markdown (hunk 9 above)
Comment (martinstibbe) [reply website/docs/r/project.html.markdown]: Fixed

File: website/docs/r/project.html.markdown (hunk 8 above)
Comment (martinstibbe) [reply website/docs/r/project.html.markdown]: Fixed

File: website/docs/r/project.html.markdown (hunk 7 above)
Comment (martinstibbe) [reply website/docs/r/project.html.markdown]: fixed

File: website/docs/d/projects.html.markdown (hunk 6 above)
Comment (martinstibbe) [reply website/docs/d/projects.html.markdown]: Fixed

File: website/docs/d/projects.html.markdown (hunk 5 above)
Comment (martinstibbe) [reply website/docs/d/projects.html.markdown]: Fixed

File: website/docs/d/project.html.markdown (hunk 4 above)
Comment (martinstibbe) [reply website/docs/d/project.html.markdown]: Fixed

File: website/docs/d/project.html.markdown (hunk 3 above)
Comment (martinstibbe) [reply website/docs/d/project.html.markdown]: Fixed

File: mongodbatlas/data_source_mongodbatlas_project.go (hunk 1 above)
Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_project.go:87]: Added tests to datasource to verify new attributes exist

**Review by andreaangiolillo — APPROVED** [review]

**Review by themantissa — APPROVED** [review]
Beautiful.  Thank you!

**Comment by wyardley** [comment]
I would argue this is, potentially, at least, breaking, and also would suggest maybe considering slightly tidier / shorter / easier to read parameter names here?

I commented more in #772

**Comment by themantissa** [comment]
@wyardley  the intended behavior was not to be breaking.   Let's continue this in #772

**Pull request merged**